controllers/constraintstatus_controller.go

```diff
@@ -0,0 +1,260 @@
+package controllers
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/open-policy-agent/gatekeeper/v3/apis/config/v1alpha1"
+	"github.com/open-policy-agent/gatekeeper/v3/apis/status/v1beta1"
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/utils/strings/slices"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var ControllerName = "contraintstatus_reconciler"
+
+type discoveryInfo struct {
+	apiResourceList        []*metav1.APIResourceList
+	discoveryLastRefreshed time.Time
+}
+
+type ConstraintStatusReconciler struct {
+	client.Client
+	Log                     logr.Logger
+	DynamicClient           dynamic.Interface
+	ClientSet               *kubernetes.Clientset
+	Scheme                  *runtime.Scheme
+	Namespace               string
```

Comment:
Is this used?

Reply:
yes	discoveryClient := r.ClientSet.Discovery() but not Scheme